### DISCUSSION TITLE
Linux Version

### DISCUSSION THREAD

**Discussion by gitalm** [Mattk70/Chirpity-Electron#56] [Show and tell]
Dear @Mattk70 ,
thanks for the wonderful software. I tested the install via **Development Setup** and it seems to be working under Linux 🐧 too.
Best regards 
![grafik](https://github.com/Mattk70/Chirpity-Electron/assets/7422149/35c17eb2-5773-4c9e-a048-221347571f37)

**Comment by Mattk70** [comment]
Hey @gitalm - it's good to hear someone has tested out the development setup instructions !!

A few notes on running Chirpity on Linux... the first point here is the main reason I haven't released an AppImage or similar for the platform.

1. I have found large batch sizes / threads values when using the CPU backend crash the application. I think this may be due to an issue with the NVIDIA driver I have installed, I don't know for sure.  _It would be good to hear if you have had this issue too?_

2. Recent releases of Chirpity have used modified NPM packages. Notably, the FFMPEG binary that is bundled in ffmpeg-static-electon does not have the audio filters for gain or normalisation compiled in. I have swapped in more capable FFMPEG builds for Windows / Mac releases. 

3. In the latest release (v1.4.0) I have made an adjustment to the timeout in processor.js of the fluent-ffmpeg package (line 492). Without this adjustment, when analysing large numbers of small files there will be numerous errors due to the issue in the file's comment:

```
            // Handle output stream events
            outputStream.target.on('close', function() {
              self.logger.debug('Output stream closed, scheduling kill for ffmpeg process');

              // Don't kill process yet, to give a chance to ffmpeg to
              // terminate successfully first  This is necessary because
              // under load, the process 'exit' event sometimes happens
              // after the output stream 'close' event.
              setTimeout(function() {
                emitEnd(new Error('Output stream closed'));
                ffmpegProc.kill();
              }, 20);
            });
```

I changed the timeout value from 20 to 100, and this seems to give enough time for the ffmpeg process to terminate properly - and fixed the errors I was getting.
